corehq/ex-submodules/soil/views.py

```diff
@@ -37,8 +37,11 @@ def heartbeat_status(request):
 
 @login_required
 def ajax_job_poll(request, download_id, template="soil/partials/dl_status.html"):
+    app_download = False
+    if 'apps' in request.META['HTTP_REFERER']:
```

Comment:
is there any way to pass a query param to the template rather than relying on this? seems a bit brittle.

also bonus points if that query param allows for a fully customizable message (either because you pass a message or we introduce some concept of message id lookups) as opposed to a boolean flag that lets you choose one of two messages.